app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/rejectedgoodsmultiple/EnterMovementReferenceNumberController.scala

```diff
@@ -0,0 +1,153 @@
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.controllers.JourneyControllerComponents
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.controllers.ReimbursementRoutes
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.controllers.SessionDataExtractor
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.controllers.SessionUpdates
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.controllers.rejectedgoodsmultiple.EnterMovementReferenceNumberController._
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.controllers.{routes => baseRoutes}
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.journeys.RejectedGoodsMultipleJourney
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.models._
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.declaration.DisplayDeclaration
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.ids.MRN
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.services.ClaimService
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.util.toFuture
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.utils.Logging
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.views.html.{rejectedgoods => pages}
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+
+@Singleton
+class EnterMovementReferenceNumberController @Inject() (
+  val jcc: JourneyControllerComponents,
+  claimService: ClaimService,
+  enterMovementReferenceNumberPage: pages.enter_movement_reference_number
+)(implicit ec: ExecutionContext, viewConfig: ViewConfig, cc: MessagesControllerComponents)
+    extends RejectedGoodsMultipleJourneyBaseController
+    with SessionDataExtractor
+    with SessionUpdates
+    with Logging {
+
+  implicit val dataExtractor: DraftClaim => Option[MRN] = _.movementReferenceNumber
+
+  def show(): Action[AnyContent] = actionReadJourney { implicit request => journey =>
+    Future.successful {
+      val emptyForm = movementReferenceNumberForm
+      val form      = journey.getLeadMovementReferenceNumber.fold(emptyForm)(emptyForm.fill)
+      val router    = ReimbursementRoutes(JourneyBindable.Multiple)
+      Ok(
+        enterMovementReferenceNumberPage(
+          form,
+          router.subKey,
+          routes.EnterMovementReferenceNumberController.enterMrnSubmit()
+        )
+      )
+    }
+  }
+
+  def enterMrnSubmit(): Action[AnyContent] = actionReadJourney { implicit request => journey =>
+    movementReferenceNumberForm
+      .bindFromRequest()
+      .fold(
+        formWithErrors =>
+          BadRequest(
+            enterMovementReferenceNumberPage(
+              formWithErrors,
+              ReimbursementRoutes(JourneyBindable.Multiple).subKey,
+              routes.EnterMovementReferenceNumberController.enterMrnSubmit()
+            )
+          ),
+        mrnNumber => {
+          val isSameAsPrevious =
+            journey.answers.movementReferenceNumbers.map(mrnSeq => mrnSeq).exists(_.contains(mrnNumber))
+
+          if (isSameAsPrevious && journey.hasCompleteAnswers)
```

Comment:
we shouldn't check for these conditions as it is now done implicitly by RejectedGoodsMultipleJourneyBaseController, you can simply call submitMovementReferenceNumberAndDeclaration and navigate to the next page, if session does not change and joureny is complete and CYA change mode it will navigate to CYA automagically.